utils/self-extr-exec/compressor.cpp

```diff
@@ -0,0 +1,352 @@
+#include <cstring>
```

Comment:
Btw, compressor can be written in C++ with exceptions, RAII, etc...
Only decompressor is intended to be lightweight.

Reply:
I made it to make the compressor and decompressor look the same. I can change it, but this version looks good to me.

Reply:
Ok.